docs/source/backends/openwrt.rst

```diff
@@ -2285,6 +2285,244 @@ Will be rendered as follows::
+| key name              | type    | default   | description                                                     |
++=======================+=========+===========+=================================================================+
+| ``interface``         | string  | ``None``  | name of the wireguard interface,                                |
+|                       |         |           |                                                                 |
+|                       |         |           | 2 to 15 alphanumeric characters, dashes and underscores         |
++-----------------------+---------+-----------+-----------------------------------------------------------------+
+| ``route_allowed_ips`` | boolean | ``False`` | automatically create a route for each Allowed IPs for this peer |
++-----------------------+---------+-----------+-----------------------------------------------------------------+
+
+WireGuard example
+~~~~~~~~~~~~~~~~~
+
+The following *configuration dictionary*:
+
+.. code-block:: python
+
+    {
+        "interfaces": [
+            {
+                "name": "wg",
+                "type": "wireguard",
+                "private_key": "QFdbnuYr7rrF4eONCAs7FhZwP7BXX/jD/jq2LXCpaXI=",
+                "port": 51820,
+                "mtu": 1420,
+                "nohostroute": false,
+                "fwmark": "",
+                "ip6prefix": [],
+                "addresses": [
+                    {
+                        "proto": "static",
+                        "family": "ipv4",
+                        "address": "10.0.0.5/32",
+                        "mask": 32,
+                    }
+                ],
+                "network": "",
+            }
+        ],
+        "wireguard_peers": [
+            {
+                "interface": "wg",
+                "public_key": "94a+MnZSdzHCzOy5y2K+0+Xe7lQzaa4v7lEiBZ7elVE=",
+                "allowed_ips": ["10.0.0.1/32"],
+                "endpoint_host": "wireguard.test.com",
+                "endpoint_port": 51820,
+                "preshared_key": "",
+                "persistent_keepalive": 60,
+                "route_allowed_ips": true,
+            }
+        ]
+    }
+
+Will be rendered as follows:
+
+.. code-block:: text
+
+    package network
+
+    config interface 'wg'
+            list addresses '10.0.0.5/32/32'
+            option listen_port '51820'
+            option mtu '1420'
+            option nohostroute '0'
+            option private_key 'QFdbnuYr7rrF4eONCAs7FhZwP7BXX/jD/jq2LXCpaXI='
+            option proto 'wireguard'
+
+    config wireguard_wg 'wgpeer'
+            list allowed_ips '10.0.0.1/32'
+            option endpoint_host 'wireguard.test.com'
+            option endpoint_port '51820'
+            option persistent_keepalive '60'
+            option public_key '94a+MnZSdzHCzOy5y2K+0+Xe7lQzaa4v7lEiBZ7elVE='
+            option route_allowed_ips '1'
+
+VXLAN over WireGuard
+--------------------
```

Comment:
I think there's a misunderstanding here.

VXLAN can work independently from wireguard, so we should document it independently.

VXLAN is a layer 2 tunnel which can be encapsulated in any L3 technology, so I don't think we need to specify VXLAN over wireguard.

Maybe, we can leave one last example showing how to encapsulate a vxlan tunnel in a wireguard tunnel.